### PR TITLE
[loki-distributed] Add a pure ingress option to use instead of the gateway service

### PR DESCRIPTION
**Pull request by aplovich**
Signed-off-by: Anthony Plovich <aplovich@anl.gov>

This adds the option to route requests using just a K8s ingress instead of the gateway service.  
https://github.com/grafana/helm-charts/issues/1459

Some things to note:
- Users will have to supply authn configuration specific to their ingress controller.
- If you're hosting Grafana on the same cluster, you'll need to point its Loki datasource at the ingress URL instead of one of the cluster's internal services.

### CODE REVIEW

File: charts/loki-distributed/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{{- if .Values.ingress.enabled }}
```
Comment (trevorwhitney) [inline charts/loki-distributed/templates/ingress.yaml:1]: should we also check to make sure the gateway is disabled here? I'd like to enforce the 1 or the other restriction in code if possible.

**Review by trevorwhitney — COMMENTED** [review]
I don't have much experience with k8s ingress, but this seems reasonable to me. I would like to enforce the 1-or-the-other gateway vs. ingress in code. The mimir [helm-chart](https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/templates/validate.yaml) does some of this stuff through a `validate` tempalte, WDYT of something like that?

File: charts/loki-distributed/templates/ingress.yaml (hunk 1 above)
Comment (aplovich) [reply charts/loki-distributed/templates/ingress.yaml:1]: Sounds like a good idea.  I'll take a look at using the validate template for this today, I haven't used one for that yet.

**Review by trevorwhitney — APPROVED** [review]
lgtm

**Pull request merged**